test/test_torch.py

```diff
@@ -5322,14 +5322,21 @@ def test_pickle_gradscaler(self, device):
         # to cuda Tensors, because I don't want to do cuda things if device is not cuda.
         # In case 3, a and b are enabled and we may also try lazy-initing _scale to a cuda tensor.
         device = torch.device(device)
-        try_lazy_inits = (True, False) if device.type == "cuda" else (False,)
+        try_lazy_inits = (True, False)
+        GradScaler = torch.cuda.amp.GradScaler if device.type == "cuda" else torch.cpu.amp.GradScaler
```

Comment:
```suggestion
        GradScaler = torch.amp.Gradscaler(device=device.type)
```
Could this be changed to a more generic device Gradscaler? Otherwise when other back ends decide to add support for Gradscaler they would need to update this test. Particularly if an out of tree backend decides to add support for Gradscaler, this might not even be possible. This feels more generalizable.

Reply:
Thanks for your comments. Modified.